### PR TITLE
chore(ui,tests): Add ARIA accessibility to the Registries page

### PR DESCRIPTION
**Pull request by xbabalov**
### What does this PR do?
Provides ARIA labels for the Registries settings page for better (testing) accessibility. 

### What issues does this PR fix or reference?
https://github.com/containers/podman-desktop/issues/4376

### How to test this PR?
yarn test

### CODE REVIEW

File: packages/renderer/src/lib/preferences/PreferencesRegistriesEditing.svelte [hunk 1]
```diff
@@ -252,7 +252,7 @@ const processPasswordElement = (node: HTMLInputElement, registry: containerDeskt
 
   <div class="container bg-charcoal-600 rounded-md p-3">
     <!-- Registries table start -->
-    <div class="w-full border-t border-b border-gray-900">
+    <div class="w-full border-t border-b border-gray-900" role="region" aria-label="Registries table">
```
Comment (odockal) [inline packages/renderer/src/lib/preferences/PreferencesRegistriesEditing.svelte]: I would use the description on ARIA table suggestions as per https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Roles/table_role.

In this div, it would be right to have also `role="table"` and then we can have `aria-label="Registries"`.

File: packages/renderer/src/lib/preferences/PreferencesRegistriesEditing.svelte [hunk 2]
```diff
@@ -252,7 +252,7 @@ const processPasswordElement = (node: HTMLInputElement, registry: containerDeskt
...
       <div class="flex w-full">
```
Comment (odockal) [inline packages/renderer/src/lib/preferences/PreferencesRegistriesEditing.svelte]: Here we could probably use `role="rowgroup"` or maybe even better, just a `row` role since there will be only one row. although, please take a look at specification, you might get better idea. Then, propably `aria-label="header"` attribute.

File: packages/renderer/src/lib/preferences/PreferencesRegistriesEditing.svelte [hunk 3]
```diff
@@ -252,7 +252,7 @@ const processPasswordElement = (node: HTMLInputElement, registry: containerDeskt
...
         <div class="flex-1 text-left py-4 pl-5 text-sm font-bold w-auto">Registry Location</div>
```
Comment (odockal) [inline packages/renderer/src/lib/preferences/PreferencesRegistriesEditing.svelte]: Here is could be a role `columnheader`. Please see the docs.

File: packages/renderer/src/lib/preferences/PreferencesRegistriesEditing.svelte [hunk 4]
```diff
@@ -261,7 +261,7 @@ const processPasswordElement = (node: HTMLInputElement, registry: containerDeskt
 
       {#each $registriesInfos as registry}
         <!-- containerDesktopAPI.Registry row start -->
-        <div class="flex flex-col w-full border-t border-gray-900">
+        <div class="flex flex-col w-full border-t border-gray-900" aria-label="{registry.serverUrl}">
```
Comment (odockal) [inline packages/renderer/src/lib/preferences/PreferencesRegistriesEditing.svelte]: For each div, that denounce the row, please use probably `row` role and `aria-label="registry.NAME?"` the property of the registry object that reference the Registry Name ( ie. "Docker Hub", "Red Hat Quay", etc.). We could then use `aria-labeledby="{registry.serverUrl}"`.

File: packages/renderer/src/lib/preferences/PreferencesRegistriesEditing.svelte [hunk 5]
```diff
@@ -294,6 +294,7 @@ const processPasswordElement = (node: HTMLInputElement, registry: containerDeskt
                   <input
                     type="text"
                     placeholder="Username"
+                    aria-label="Username"
```
Comment (odockal) [inline packages/renderer/src/lib/preferences/PreferencesRegistriesEditing.svelte:300]: We could also probably use `role="cell"` on a proper level - probably on the div that is the most outer for given cell.

Comment (odockal) [reply packages/renderer/src/lib/preferences/PreferencesRegistriesEditing.svelte:300]: This also counts for other cells we need to operate over.

File: packages/renderer/src/lib/ui/DropdownMenu.svelte [hunk 6]
```diff
@@ -41,7 +41,8 @@ function onWindowClick(e: any) {
       toggleMenu();
     }}"
     bind:this="{outsideWindow}"
-    class="mr-2 text-gray-400 hover:bg-charcoal-800 hover:text-purple-400 font-medium rounded-md inline-flex items-center px-2 py-2 text-center">
+    class="mr-2 text-gray-400 hover:bg-charcoal-800 hover:text-purple-400 font-medium rounded-md inline-flex items-center px-2 py-2 text-center"
+    aria-label="Dropdown menu">
```
Comment (odockal) [inline packages/renderer/src/lib/ui/DropdownMenu.svelte]: How is this change related to registries?

**Review by odockal — CHANGES_REQUESTED** [review]

File: packages/renderer/src/lib/ui/DropdownMenu.svelte (hunk 6 above)
Comment (xbabalov) [reply packages/renderer/src/lib/ui/DropdownMenu.svelte]: For each registry row there's a dropdown menu to offer the edit and delete options.

File: packages/renderer/src/lib/preferences/PreferencesRegistriesEditing.svelte (hunk 4 above)
Comment (xbabalov) [reply packages/renderer/src/lib/preferences/PreferencesRegistriesEditing.svelte]: The newly added registries dont have the name attribute but are labeled by their url, so I kept serverUrl for the new registries and name for default registries. I can further separate by adding another rowgroup in between the new and default registries if that's desired.

File: packages/renderer/src/lib/ui/DropdownMenu.svelte (hunk 6 above)
Comment (odockal) [reply packages/renderer/src/lib/ui/DropdownMenu.svelte]: @xbabalov ok, thanks!

File: packages/renderer/src/lib/preferences/PreferencesRegistriesEditing.svelte (hunk 4 above)
Comment (odockal) [reply packages/renderer/src/lib/preferences/PreferencesRegistriesEditing.svelte]: I would not introduce another separator, instead, can we use ternary operator and use name if the property exists and url if it does not?

**Review by odockal — APPROVED** [review]
